ddl/ddl_api.go

```diff
@@ -1095,6 +1095,59 @@ func checkConstraintNames(constraints []*ast.Constraint) error {
 	return nil
 }
 
+func setTableAutoShardBits(tbInfo *model.TableInfo, colDefs []*ast.ColumnDef) error {
+	allowAutoShard := config.AllowExperimentalAutoShard()
+	pkColName := tbInfo.GetPkName()
+	return forEachColWithAutoShardOpt(colDefs, func(col *ast.ColumnDef, autoShardBits uint64) error {
+		if !allowAutoShard {
+			return ErrInvalidAutoShard.GenWithStackByArgs(autoid.AutoShardExperimentalDisabledErrMsg)
+		}
+
+		if tbInfo.PKIsHandle && col.Name.Name.L == pkColName.L {
```

Comment:
According to https://github.com/pingcap/tidb/blob/153a6facfef50c11c7884a6a90f6cdb316f904cb/ddl/ddl_api.go#L1137 , only tables created with config `alter-primary-key = false` will set the PKIsHandle as true.  So this feature is not available with config `alter-primary-key = true`?

Reply:
Yes, it's not available when `PKIsHandle = false`, we can use `shard_row_id_bits` instead to achieve the same purpose in this case.